decoder-browser.js

```diff
@@ -6,18 +6,22 @@ function defaultDecoder(data) {
   return out.replace(/\0/g, '').trim();
 }
 module.exports = createDecoder;
-var regex = /^(?:ASNI\s)?(\d+)$/m;
+var regex = /^(?:ANSI\s)?(\d+)$/m;
+
 function createDecoder(encoding) {
   if (!encoding) {
     return defaultDecoder;
   }
+  enconding = String(encoding).trim();
```

Comment:
do we just need to check if it's a buffer?

Reply:
What can we do if it's toString method doesn't decode it properly, as the Uint8Array buffer in the new test? Then we need to know the encoding of the encoding? 😆 The worst thing that could happen I guess is that we get something like `"[object Buffer]"` and `TextDecoder` throws because of invalid encoding label, but then we get the default decoder. Or do you think about something else?

I just saw that I made a typo in that very line though! 😳